test/phlex/valueable_object.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+describe Phlex::HTML do
+	def build_component_with_template(&block)
+		Class.new(Phlex::HTML) do
+			define_method(:view_template, block)
+		end
+	end
+
+	it "renders to_s class" do
+		klass = Class.new { define_method(:to_s, -> { "to_s" }) }
+		component = build_component_with_template do
+			div class: klass.new
+		end
+
+		expect(component.call).to be == %(<div class="to_s"></div>)
+	end
+
+	it "renders to_str class" do
+		klass = Class.new { define_method(:to_str, -> { "to_str" }) }
+		component = build_component_with_template do
+			div class: klass.new
+		end
+
+		expect(component.call).to be == %(<div class="to_str"></div>)
+	end
+
+	it "renders to_phlex_attribute_value class" do
+		klass = Class.new { define_method(:to_phlex_attribute_value, -> { "to_phlex_attribute_value" }) }
+		component = build_component_with_template do
+			div class: klass.new
+		end
+
+		expect(component.call).to be == %(<div class="to_phlex_attribute_value"></div>)
+	end
+
+	it "renders phlex attribute value according to method call precedence" do
+		to_phlex_attribute_value_klass = Class.new { define_method(:to_phlex_attribute_value, -> { "to_phlex_attribute_value" }) }
+		to_str_klass = Class.new(to_phlex_attribute_value_klass) { define_method(:to_str, -> { "to_str" }) }
+		to_s_klass = Class.new(to_str_klass) { define_method(:to_s, -> { "to_s" }) }
```

Comment:
I think this would need to be just one class with all three methods defined to ensure the `to_phlex_attribute` method “wins”.

Reply:
Thanks for the feedback @joeldrapper!

I refactored the test so that there is only one class that defines all three methods to ensure correct method call precedence